lib/charms/observability_libs/v0/cert_handler.py

```diff
@@ -0,0 +1,343 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""## Overview.
+
+This document explains how to use the `CertHandler` class to
+create and manage TLS certificates through the `tls_certificates` interface.
+
+The goal of the CertHandler is to provide a wrapper to the `tls_certificates`
+library functions to make the charm integration smoother.
+
+## Library Usage
+
+This library should be used to create a `CertHandler` object, as per the
+following example:
+
+```python
+cert_handler = CertHandler(
+    charm=self,
+    peer_relation_name="replicas",
+    cert_subject="unit_name",  # Optional
+    key="cert-manager"  # Optional
+)
+```
+
+This library requires a peer relation to be declared in the requirer's metadata. Peer relation data
+is used for "persistent storage" of the private key and certs.
+"""
+import socket
+from typing import Optional, Union, List
+import json
+
+try:
+    from charms.tls_certificates_interface.v2.tls_certificates import (  # pyright: ignore
+        AllCertificatesInvalidatedEvent,
+        CertificateAvailableEvent,
+        CertificateExpiringEvent,
+        CertificateInvalidatedEvent,
+        TLSCertificatesRequiresV2,
+        generate_csr,
+        generate_private_key,
+    )
+except ImportError:
+    raise ImportError(
+        "charms.tls_certificates_interface.v2.tls_certificates is missing; please get it through charmcraft fetch-lib"
+    )
+from ops.charm import CharmBase, RelationBrokenEvent
+from ops.model import Relation
+from ops.framework import EventBase, EventSource, Object, ObjectEvents
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+LIBID = "deadbeef"
```

Comment:
we probably need to publish it and get a libid before we merge

Reply:
Done.